include/dmlc/logging.h

```diff
@@ -251,26 +251,39 @@ class CustomLogMessage {
   std::ostringstream log_stream_;
 };
 
+
+
+#if DMLC_LOG_STACK_TRACE
+std::string demangle(char const* msg_str);
+#define STACK_TRACE()\
```

Comment:
why use macro instead of function?

Reply:
I can use a function, then it will show on the first position in the stack trace. I would prefer a function, is more clean. Would you prefer it that way?